model_rnn/adapter.py

```diff
@@ -33,26 +36,16 @@ def _predict_helper(self, title:str, top_n:int=10, max_length:int=100, result_li
                 topv, topi = output.reshape(-1).topk(2)
                 top_char = IX_TO_CHAR[topi[0].item()]
                 top_2_char = IX_TO_CHAR[topi[1].item()]
+
+                if n > 0 and top_2_char != EOS:
+                    n = n // 2
+                    for result in self._predict_helper(title + top_2_char, n, max_length, score + topv[1].item()):
```

Comment:
This approach can be optimized better. Once you vectorize user prefix, you don't ever need to vectorize it again. It's not what you are doing here through.